### DISCUSSION TITLE
Can I have two instances of inercia js on one page?

### DISCUSSION THREAD

**Discussion by Stevemoretz** [inertiajs/inertia#796] [Help] [answered]
Well you may think I'm crazy but it's a really common use case:

```
Some html
{inertia js here to load a react component}
Some html
Some html
{inertia js here to load a react component} 
```

If I could somehow use it like this, I could get rid of the stupid web components.

**Comment by ajnsn** [comment]
@Stevemoretz Inertia is more like an SPA. The main difference is the that there's no client-side routing. While multiple instances might be achievable somehow, there's no direct support for it.

**Reply by Stevemoretz** [reply]
Hi thanks you so much for the reply, a couple of questions, you mentioned "no client-side routing" I thought it has client side routing unless we reload the page manually it was mentioned in the docs or am I getting it wrong?

Secondly I realize now my request is kind of stupid, if inercia is built with the purpose of spa in mind. It can't be changed to webcomponent so easily, but I think it could be great if you considered it for the future.

One of the reasons that would be useful is for seo and another for combining technologies.

Sounds like you got seo covered with server-side rendering, when it approximately will be out of feature preview phase though?

**Comment by ajnsn** [comment, marked as answer]
@Stevemoretz Inertia always does a server round-trip. What it prevents is a full page reload. When you are using Laravel as a backend, the Livewire approach might be a better fit. 

Regarding the SSR, there's a already a preview available. Atm you need to [sponsor Jonathan](https://inertiajs.com/sponsors#become-a-sponsor) to access it.

**Reply by Stevemoretz** [reply]
Sounds great thanks.
